### PR TITLE
Feat ratings

### PR DESCRIPTION
**Pull request by mikeyrayvon**
adds ratings feature. lists average ratings for admin. adds rating buttons for committee on single submission review

### CODE REVIEW

**Comment by mikeyrayvon** [comment]
to see this functionality: override afterDeadline helper to return true

**Comment by mixcas** [comment]
@mikeyrayvon i had checked your code and it's all fine. i was going to merge but then i made some changes
can u check those changes and merge?

**Comment by mixcas** [comment]
actually @varffs should check it all

File: app/client/css/main.styl [hunk 1]
```diff
@@ -18,6 +18,9 @@ $color-grey = rgb(158,158,158)
 html
   background-color: $color-white-soft
   color: $color-black-soft
+  
+.navy
```
Comment (varffs) [inline app/client/css/main.styl:22]: this isnt a great class name. Should be background-blue I think

File: app/client/css/main.styl [hunk 2]
```diff
@@ -18,6 +18,9 @@ $color-grey = rgb(158,158,158)
...
+  background-color: $color-blue !important
```
Comment (varffs) [inline app/client/css/main.styl:23]: Why !important?

File: app/client/templates/admin/submissionReview.html [hunk 3]
```diff
@@ -74,6 +81,43 @@
       </div>
     </div>
   </div>
-{{/with}}
+  {{#if afterDeadline}}
```
Comment (varffs) [inline app/client/templates/admin/submissionReview.html]: better to check if application is submitted here

File: app/server/methods/ratingsMethods.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
+Meteor.methods({
+  rateApplication: function(ratingNumber, applicationId) {
+    check(ratingNumber, Number);
+    check(applicationId, String);
+
+    if (!Meteor.userId()) {
```
Comment (varffs) [inline app/server/methods/ratingsMethods.js]: need to migrate this here to `this.userId` I think

File: app/server/methods/ratingsMethods.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+      throw new Meteor.Error('error-not-signed-in', 'You must sign in first.');
+    }
+
```
Comment (varffs) [inline app/server/methods/ratingsMethods.js:9]: No permissions check here! Need to check if is committee role

File: app/client/css/main.styl (hunk 1 above)
Comment (varffs) [inline app/client/css/main.styl:22]: ergh this is materialize this sucks

File: app/client/templates/admin/submissions.html [hunk 6]
```diff
@@ -1,11 +1,17 @@
 <template name="submissions">
   <div class="container">
-    <table class="striped js-sortable" data-sortlist="[[3,1]]">
+    <table class="striped js-sortable" data-sortlist="[[2,1]]">
       <thead>
         <tr>
+            <th data-field="folio">{{_ "application.folio.label"}}</th>
             <th data-field="name">{{_ "application.name.label"}}</th>
-            <th data-field="project">{{_ "application.projectTitle.label"}}</th>
+            <th data-field="project" data-empty="bottom">{{_ "application.projectTitle.label"}}</th>
```
Comment (varffs) [inline app/client/templates/admin/submissions.html:8]: for reference @mikeyrayvon what does that data-empty thing do?

Comment (mixcas) [inline app/client/templates/admin/submissions.html:8]: @varffs it tells the sorting system what the 'default' value is when its empty
ex, in line 11 `data-empty="zero"`, when rating is empty the sorting will assume is zero

File: app/server/methods/ratingsMethods.js [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+    if (!Roles.userIsInRole(this.userId, ['committee',])) {
+      throw new Meteor.Error('not-allowed', 'You must be a committee user aka No Juice Error');
+    }
+
+    // Find existing rating
+    var existingRating = Ratings.findOne({userId: Meteor.userId(), applicationId: applicationId,});
+    
+    // Create rating update
+    var rating = {
```
Comment (mixcas) [inline app/server/methods/ratingsMethods.js]: We could get rid of `rating` object and update `existingRating` with new values instead, right?

**Comment by mixcas** [comment]
This is probably the PR that has took the longest…
I made some updates, read everything like 4 times already and everything seems fine. 
Can you merge, @varffs ?

File: app/client/templates/admin/submissions.html (hunk 6 above)
Comment (varffs) [inline app/client/templates/admin/submissions.html:8]: but zero as in a string?

Comment (mixcas) [inline app/client/templates/admin/submissions.html:8]: oh… umm… idk… @mikeyrayvon ?

**Pull request merged**